### DISCUSSION TITLE
Explanation regarding Inference Results

### DISCUSSION THREAD

**Discussion by sunny52266** [opengeos/geoai#89] [Q&A] [answered]
Hi!

First of all, I would like to thank you for your work. 

Currently, I am testing to train a model for parcel mapping purposes (not building extraction).

The input training data contains 5cm imagery of a small area and its relevant parcels digitized in shapefile format. (also tried geojson). 

The vector dataset includes a class field. 

following is a sample of the input dataset:

![image](https://github.com/user-attachments/assets/60cb0169-c34a-4f70-a477-5e54c1b5558e)

I tried training the model on 2, 10 and 50 epochs. 

However, the resulting inference seems to segment the whole block instead of individual polygons. like the following:

![image](https://github.com/user-attachments/assets/ab930446-c926-47bb-bf60-55665773aa8e)

Is there something I am missing or is it supposed to be like this?

**Comment by giswqs** [comment, marked as answer]
For semantic segmentation, adjacent objects in the binary masks are merged together when converting to polygons. This issue will be resolved when instance segmentation is supported.
